File: taurus-be/src/main/java/com/fundaro/zodiac/taurus/service/UsersService.java
package com.fundaro.zodiac.taurus.service;

import com.fundaro.zodiac.taurus.domain.Users;
import com.fundaro.zodiac.taurus.domain.criteria.UserCalendarEventsCriteria;
import com.fundaro.zodiac.taurus.domain.criteria.UsersCriteria;
import com.fundaro.zodiac.taurus.service.dto.CalendarEventsDTO;
import com.fundaro.zodiac.taurus.service.dto.UsersDTO;
import com.fundaro.zodiac.taurus.service.dto.UsersMeDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.security.authentication.AbstractAuthenticationToken;

import java.util.Optional;

/**
 * Service Interface for managing {@link Users}.
 */
public interface UsersService extends CommonOpenSearchService<Users, UsersDTO, UsersCriteria> {
    Optional<UsersDTO> findMe(AbstractAuthenticationToken abstractAuthenticationToken);

    UsersDTO partialUpdateOwn(UsersMeDTO dto, AbstractAuthenticationToken abstractAuthenticationToken);

    void deleteOwn(AbstractAuthenticationToken abstractAuthenticationToken);

    void deleteOwnForGdpr(AbstractAuthenticationToken abstractAuthenticationToken);

    void deleteForGdpr(Long id, AbstractAuthenticationToken abstractAuthenticationToken);

    void sendSetupEmail(Long id, AbstractAuthenticationToken abstractAuthenticationToken);

    Page<CalendarEventsDTO> getUserCalendarEvents(Long id, UserCalendarEventsCriteria criteria, Pageable pageable, AbstractAuthenticationToken abstractAuthenticationToken);

    Page<CalendarEventsDTO> getCurrentUserCalendarEvents(UserCalendarEventsCriteria userCalendarEventsCriteria, Pageable pageable, AbstractAuthenticationToken abstractAuthenticationToken);
}

File: taurus-fe/src/app/service/users.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { CalendarEvents, Page, Users, UsersCalendarEventsCriteria, UsersCriteria } from '../module';
import { CommonOpenSearchService } from './common-open-search.service';

export interface UsersMeDTO {
    name?: string;
    lastName?: string;
    email?: string;
}

@Injectable({
    providedIn: 'root'
})
export class UsersService extends CommonOpenSearchService<Users, UsersCriteria> {
    override resourceName(): string {
        return "users";
    }

    public getOwn(): Observable<Users> {
        return this.http.get<Users>(`${this.baseUrl}/${this.resourceName()}/me`);
    }

    public getOwnCalendarEvents(criteria?: UsersCalendarEventsCriteria): Observable<Page<CalendarEvents>> {
        const options = this.createRequestOption(criteria);
        return this.http.get<Page<CalendarEvents>>(`${this.baseUrl}/${this.resourceName()}/me/calendar-events`, { params: options, observe: 'body' });
    }

    public partialUpdateOwn(dto: UsersMeDTO): Observable<Users> {
        return this.http.patch<Users>(`${this.baseUrl}/${this.resourceName()}/me`, dto);
    }

    public deleteOwn(): Observable<void> {
        return this.http.delete<void>(`${this.baseUrl}/${this.resourceName()}/me`);
    }

    public deleteOwnForGdpr(): Observable<void> {
        return this.http.delete<void>(`${this.baseUrl}/${this.resourceName()}/me/gdpr`);
    }

    public deleteForGdpr(id: string): Observable<void> {
        return this.http.delete<void>(`${this.baseUrl}/${this.resourceName()}/${id}/gdpr`);
    }

    public sendSetupEmail(id: string): Observable<void> {
        return this.http.put<void>(`${this.baseUrl}/${this.resourceName()}/${id}/send-setup-email`, null);
    }

    public getUserCalendarEvents(id: string, criteria?: UsersCalendarEventsCriteria): Observable<Page<CalendarEvents>> {
        const options = this.createRequestOption(criteria);
        return this.http.get<Page<CalendarEvents>>(`${this.baseUrl}/${this.resourceName()}/${id}/calendar-events`, { params: options, observe: 'body' });
    }
}
